pkg/plugin/v1/init.go

```diff
@@ -0,0 +1,169 @@
+
+Writes the following files:
+- a helm-charts directory with the chart(s) to build releases from
+- a watches.yaml file that defines the mapping between your API and a Helm chart
+- a PROJECT file with the domain and repo
+- a Makefile to build the project
+- a Kustomization.yaml for customizating manifests
+- a Patch file for customizing image for manager manifests
+- a Patch file for enabling prometheus metrics
+`
+	ctx.Examples = fmt.Sprintf(`  $ %s init --plugins=%s \
+      --domain=example.com \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService \
+      --helm-chart=myrepo/app
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=myrepo/app
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=myrepo/app \
+      --helm-chart-version=1.2.3
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/ \
+      --helm-chart-version=1.2.3
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=/path/to/local/chart-directories/app/
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
+`,
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+	)
+
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.SortFlags = false
+	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
+	p.apiPlugin.BindFlags(fs)
+}
+
+func (p *initPlugin) InjectConfig(c *config.Config) {
+	// v3 project configs get a 'layout' value.
+	c.Layout = plugin.KeyFor(Plugin{})
+	p.config = c
+}
+
+func (p *initPlugin) Run() error {
+	return cmdutil.Run(p)
+}
+
+func (p *initPlugin) Validate() error {
+	// Check if the project name is a valid namespace according to k8s
+	dir, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("error to get the current path: %v", err)
+	}
+	projectName := filepath.Base(dir)
+	if err := validation.IsDNS1123Label(strings.ToLower(projectName)); err != nil {
+		return fmt.Errorf("project name (%s) is invalid: %v", projectName, err)
+	}
+
+	defaultOpts := chartutil.CreateOptions{CRDVersion: "v1"}
+	if !p.apiPlugin.gvk.Empty() || p.apiPlugin.createOptions != defaultOpts {
+		p.doAPIScaffold = true
+		return p.apiPlugin.Validate()
+	}
+	return nil
```

Comment:
We need to hav the operator name in the PROJECT file because It is used by SDK features (bundle)

```suggestion
       // set the repo to provide the operator name
	if p.config.Repo == "" {
		p.config.Repo = projectName
	} 
	return nil
```